x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/ExpressionTests.java

```diff
@@ -136,6 +137,60 @@ public void testBackQuotedAttribute() {
     }
 
+    public void testBackQuotedIdentifierWithEscapedBackQuote() {
+        String quote = "`";
+        String qualifier = "``test``table``";
+        String expectedQualifier = "`test`table`";
+        String name = "@timestamp";
+        Expression exp = expr(quote + qualifier + quote + "." + quote + name + quote);
+        assertThat(exp, instanceOf(UnresolvedAttribute.class));
+        UnresolvedAttribute ua = (UnresolvedAttribute) exp;
+        assertThat(ua.name(), equalTo(expectedQualifier + "." + name));
+        assertThat(ua.qualifiedName(), equalTo(expectedQualifier + "." + name));
+        assertThat(ua.qualifier(), is(nullValue()));
+
+        quote = "`";
+        qualifier = "``test_table";
+        expectedQualifier = "`test_table";
+        name = "@timestamp";
+        exp = expr(quote + qualifier + quote + "." + quote + name + quote);
+        assertThat(exp, instanceOf(UnresolvedAttribute.class));
+        ua = (UnresolvedAttribute) exp;
+        assertThat(ua.name(), equalTo(expectedQualifier + "." + name));
+        assertThat(ua.qualifiedName(), equalTo(expectedQualifier + "." + name));
+        assertThat(ua.qualifier(), is(nullValue()));
+    }
+
+    public void testBackQuotedIdentifierWithUnescapedBackQuotes() {
+        ParsingException e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("`wrong_identifier == true"));
+        assertEquals("line 1:1: token recognition error at: '`wrong_identifier == true'", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("``wrong_identifier == true"));
+        assertThat(e.getMessage(), startsWith("line 1:3: mismatched input 'wrong_identifier' expecting {<EOF>, "));
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("``wrong_identifier` == true"));
+        assertEquals("line 1:19: token recognition error at: '` == true'", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("`wrong`identifier` == true"));
+        assertEquals("line 1:18: token recognition error at: '` == true'", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("wrong_identifier` == true"));
+        assertEquals("line 1:17: token recognition error at: '` == true'", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("wrong_identifier`` == true"));
+        assertThat(e.getMessage(), startsWith("line 1:17: mismatched input '``' expecting {<EOF>,"));
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("`wrong_identifier`` == true"));
+        assertEquals("line 1:19: token recognition error at: '` == true'", e.getMessage());
+    }
```

Comment:
This means that any identifier containing a backquote needs quoting.
/cc @jrodewig